### PR TITLE
Public URL with Checkin Statistics

### PR DESCRIPTION
**Pull request by salimhb**
https://trello.com/c/yg81s70e

Public URL `/companies/<company_id>/stats`

Returns
```
[
  {
    "area_name": "Bedroom",
    "checkin_count": 3
  },
  {
    "area_name": "test",
    "checkin_count": 1
  }
]
```

### CODE REVIEW

File: app/controllers/owners/companies_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module Owners
   class CompaniesController < Owners::ApplicationController
-    before_action :authenticate_owner!
+    before_action :authenticate_owner!, except: %i[stats]
```
Comment (bonflintstone) [inline app/controllers/owners/companies_controller.rb:3]: So I think we should have some kind of authentication, one possible solution was to give them a custom token we set in the admin, or perhaps to use the normal authentication. But completely open is not really a good idea for all the other companies I believe

File: app/controllers/owners/companies_controller.rb [hunk 2]
```diff
@@ -28,6 +28,16 @@ def show
       render json: company
     end
 
+    def stats
+      company = Company.find(params[:company_id])
+
+      stats = company.areas.map do |a|
+        { area_name: a.name, checkin_count: a.tickets.open.count }
```
Comment (bonflintstone) [inline app/controllers/owners/companies_controller.rb:35]:
```suggestion
      stats = company.areas.map do |area|
        { area_name: area.name, checkin_count: area.tickets.open.count }
```
I really prefer non-single-char variable names :smile:

**Review by bonflintstone — CHANGES_REQUESTED** [review]

File: config/credentials.yml.enc [hunk 3]
```diff
@@ -0,0 +1 @@
+xoiesKdGgN4GkiHNrLNqhSbEjSp9stspi722YKgaA1IcNEH1ZsAlR4QDDcda9putqlbA4zKulyo3oVb1UzzShh5Rop46hGvWw/EyFyda9OcqYgcQPhMj6n2LWHD81cnwKba46lQqXcUC0kRaCXD2fwd5rtjYa/3G9CbqiWfDi0uAPVD01dAEfss1TP4W3G9QJXyCpi39LBKXCetkDPHe4uCnnMthKp5ouBZWO9NUA1EgPv3rEOMgAYYrqUNS6OPCxKKpEc39ZhNUs2ZEtUVGWDquiLZflqQgMp3jKHDylbIAmZ13eUFcvFi8pTsclT6QjPYKP7KufCRe/B2ipzY1/r07AOuCTT+Ds966N4vQuQ4ZPshZ6u8A+1V3h8+ZwPKzL6BLdpIIyN7fHyghSgugMlyq0Vxr1TZn16vh--1EFQmjaXgzTL+Tzf--KITWiU8ZatF/6x/pRdsijw==
```
Comment (salimhb) [inline config/credentials.yml.enc:1]: Turns out that secret_key_base is needed to create migrations 😄

File: spec/requests/owners/companies_request_spec.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-RSpec.describe Owners::CompaniesController do
+RSpec.describe Owners::CompaniesController, type: :request do
```
Comment (bonflintstone) [inline spec/requests/owners/companies_request_spec.rb:3]: Is it not configured to automatically infer the type from the folder?

**Review by bonflintstone — APPROVED** [review]
Super nice :+1:

File: spec/requests/owners/companies_request_spec.rb (hunk 4 above)
Comment (salimhb) [reply spec/requests/owners/companies_request_spec.rb:3]: It should, but I kept getting an error when I accessed `headers` below!

**Review by endverbraucher — APPROVED** [review]
Looks good to me!

File: spec/requests/owners/companies_request_spec.rb (hunk 4 above)
Comment (salimhb) [reply spec/requests/owners/companies_request_spec.rb]: welp, I tried again and it worked!

**Pull request merged**